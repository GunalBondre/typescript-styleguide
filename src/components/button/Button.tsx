import React, { FC } from 'react';
import { ButtonContainer } from './Button.style';
import { ButtonProps } from './Button.types';

const Button: FC<ButtonProps> = ({
	variant,
	color,
	size,
	children,
	borderRadius,
	noBorder,
	width,
	height,
	iconName,
	iconColor,
	fontSize,
	disabled,
	...props
}) => {
	return (
		<ButtonContainer
			variant={variant}
			color={color}
			width={width}
			height={height}
			customRadius={borderRadius}
			noBorder={noBorder}
			size={size}
			fontSize={fontSize}
			disabled={disabled}
			data-testid='button-test'
			{...props}
		>
			{children}
		</ButtonContainer>
	);
};

export default Button;
