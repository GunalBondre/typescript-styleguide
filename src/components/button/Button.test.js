import { render, screen } from '@testing-library/react';
import Button from './Button';

describe('Component Button:', () => {
	it('should render the button component', () => {
		const { container } = render(<Button variant='primary' />);

		expect(screen.getByTestId('button-test')).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});

	it('should render the button with width prop', () => {
		render(<Button variant='primary' width='350px' />);

		expect(screen.getByTestId('button-test')).toHaveStyle('width:350px');
	});

	it('should render the button with custom border radius ', () => {
		render(<Button variant='primary' customRadius='20px' />);

		expect(screen.getByTestId('button-test')).toHaveStyle('border-radius:20px');
	});

	it('should render the button with custom font size and color ', () => {
		render(<Button variant='primary' fontSize='18px' color='red' />);

		expect(screen.getByTestId('button-test')).toHaveStyle('font-size:18px');
		expect(screen.getByTestId('button-test')).toHaveStyle('color:red');
	});
});
