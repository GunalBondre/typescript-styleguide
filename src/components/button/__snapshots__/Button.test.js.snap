// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Button: should render the button component 1`] = `
<div>
  <button
    class="sc-bczRLJ kPObXD"
    data-testid="button-test"
  />
</div>
`;
